# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../cloop/gcc34.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -ur cloop-2.01/advancecomp-1.9_create_compressed_fs/7z/AriBitCoder.h cloop-2.01-fixed/advancecomp-1.9_create_compressed_fs/7z/AriBitCoder.h
--- cloop-2.01/advancecomp-1.9_create_compressed_fs/7z/AriBitCoder.h	2003-02-10 20:25:06.000000000 +0100
+++ cloop-2.01-fixed/advancecomp-1.9_create_compressed_fs/7z/AriBitCoder.h	2004-10-15 23:07:37.565529344 +0200
@@ -51,13 +51,13 @@
 public:
   void Encode(CRangeEncoder *aRangeEncoder, UINT32 aSymbol)
   {
-    aRangeEncoder->EncodeBit(m_Probability, kNumBitModelTotalBits, aSymbol);
+    aRangeEncoder->EncodeBit(this->m_Probability, kNumBitModelTotalBits, aSymbol);
     UpdateModel(aSymbol);
   }
   UINT32 GetPrice(UINT32 aSymbol) const
   {
     return g_PriceTables.m_StatePrices[
-      (((m_Probability - aSymbol) ^ ((-(int)aSymbol))) & (kBitModelTotal - 1)) >> kNumMoveReducingBits];
+      (((this->m_Probability - aSymbol) ^ ((-(int)aSymbol))) & (kBitModelTotal - 1)) >> kNumMoveReducingBits];
   }
 };
 
@@ -68,11 +68,11 @@
 public:
   UINT32 Decode(CRangeDecoder *aRangeDecoder)
   {
-    UINT32 aNewBound = (aRangeDecoder->m_Range >> kNumBitModelTotalBits) * m_Probability;
+    UINT32 aNewBound = (aRangeDecoder->m_Range >> kNumBitModelTotalBits) * this->m_Probability;
     if (aRangeDecoder->m_Code < aNewBound)
     {
       aRangeDecoder->m_Range = aNewBound;
-      m_Probability += (kBitModelTotal - m_Probability) >> aNumMoveBits;
+      this->m_Probability += (kBitModelTotal - this->m_Probability) >> aNumMoveBits;
       if (aRangeDecoder->m_Range < kTopValue)
       {
         aRangeDecoder->m_Code = (aRangeDecoder->m_Code << 8) | aRangeDecoder->m_Stream.ReadByte();
@@ -84,7 +84,7 @@
     {
       aRangeDecoder->m_Range -= aNewBound;
       aRangeDecoder->m_Code -= aNewBound;
-      m_Probability -= (m_Probability) >> aNumMoveBits;
+      this->m_Probability -= (this->m_Probability) >> aNumMoveBits;
       if (aRangeDecoder->m_Range < kTopValue)
       {
         aRangeDecoder->m_Code = (aRangeDecoder->m_Code << 8) | aRangeDecoder->m_Stream.ReadByte();
